ddl/ddl.go

```diff
@@ -1014,6 +1014,15 @@ func (d *ddl) AddColumn(ctx context.Context, ti ast.Ident, spec *ast.AlterTableS
 		return errors.Trace(err)
 	}
 
+	if mysql.HasNotNullFlag(col.Flag) && col.DefaultValue == nil {
+		zeroValue := table.GetZeroValue(&col.ColumnInfo)
+		col.DefaultValue, err = zeroValue.ToString()
```

Comment:
Seems mysql does not behave this way. It will not automatically add default value to column definition.